### ISSUE TITLE
OpenZFS does not compile on amd64 FreeBSD 15.0 (CURRENT)

### ISSUE THREAD

**Issue by mcmilk** [openzfs/zfs#16328]
### System information

Type | Version/Name
 --- | ---
Distribution Name	|FreeBSD 15
Distribution Version	|15.0-CURRENT
Kernel Version	|main-n271031-cf2c79915a63
Architecture	|amd64
OpenZFS Version	|master

### Describe the problem you're observing

Running these commands on OpenZFS master branch:
```
export MAKE="gmake"
./autogen.sh
./configure \
    --prefix=/usr/local \
    --with-libintl-prefix=/usr/local \
    --enable-pyzfs \
    --enable-debug \
    --enable-debuginfo
gmake -j`sysctl -n hw.ncpu`
```

will fail while building, with this error:

```
cc  -O2 -pipe  -std=gnu99 -Wno-format-zero-length -nobuiltininc -idirafter /usr/lib/clang/18/include -fstack-protector-strong -Wall -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wcast-qual -Wundef -Wno-pointer-sign -D__printf__=__freebsd_kprintf__ -Wmissing-include-dirs -fdiagnostics-show-option -Wno-unknown-pragmas -Wswitch -Wno-error=tautological-compare -Wno-error=empty-body -Wno-error=parentheses-equality -Wno-error=unused-function -Wno-error=pointer-sign -Wno-error=shift-negative-value -Wno-address-of-packed-member -Wno-format-zero-length   -Qunused-arguments     -include /usr/home/zfs/x/zfs/include/os/freebsd/spl/sys/ccompile.h -D__KERNEL__ -DFREEBSD_NAMECACHE -DBUILDING_ZFS -D__BSD_VISIBLE=1 -DHAVE_UIO_ZEROCOPY -DWITHOUT_NETDUMP -D__KERNEL -D_SYS_CONDVAR_H_ -D_SYS_VMEM_H_ -DKDTRACE_HOOKS -DCOMPAT_FREEBSD11 -D__x86_64 -DHAVE_SSE2 -DHAVE_SSSE3 -DHAVE_SSE4_1 -DHAVE_SSE4_2 -DHAVE_AVX -DHAVE_AVX2 -DHAVE_AVX512F -DHAVE_AVX512VL -DHAVE_AVX512BW -DZFS_DEBUG -g -DINVARIANTS -DWITNESS -DOPENSOLARIS_WITNESS -DBITS_PER_LONG=64 -fno-strict-aliasing -Werror -D_KERNEL -DKLD_MODULE -nostdinc  -I/usr/home/zfs/x/zfs/include -I/usr/home/zfs/x/zfs/module/icp/include -I/usr/home/zfs/x/zfs/include/os/freebsd -I/usr/home/zfs/x/zfs/include/os/freebsd/spl -I/usr/home/zfs/x/zfs/include/os/freebsd/zfs -I/usr/home/zfs/x/zfs/module/zstd/include -I/usr/home/zfs/x/zfs/module -include /usr/home/zfs/x/zfs/module/opt_global.h -I. -I/usr/src/sys -I/usr/src/sys/contrib/ck/include -fno-common -g -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fdebug-prefix-map=./machine=/usr/src/sys/amd64/include -fdebug-prefix-map=./x86=/usr/src/sys/x86/include -fdebug-prefix-map=./i386=/usr/src/sys/i386/include     -MD  -MF.depend.spl_policy.o -MTspl_policy.o -mcmodel=kernel -mno-red-zone -mno-mmx -mno-sse -msoft-float  -fno-asynchronous-unwind-tables -ffreestanding -fwrapv -fstack-protector -gdwarf-4 -Wall -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wcast-qual -Wundef -Wno-pointer-sign -D__printf__=__freebsd_kprintf__ -Wmissing-include-dirs -fdiagnostics-show-option -Wno-unknown-pragmas -Wswitch -Wno-error=tautological-compare -Wno-error=empty-body -Wno-error=parentheses-equality -Wno-error=unused-function -Wno-error=pointer-sign -Wno-error=shift-negative-value -Wno-address-of-packed-member -Wno-format-zero-length   -Qunused-arguments -mno-aes -mno-avx  -std=gnu99 -c /usr/home/zfs/x/zfs/module/os/freebsd/spl/spl_policy.c -o spl_policy.o
In file included from /usr/home/zfs/x/zfs/module/os/freebsd/spl/spl_policy.c:37:
In file included from /usr/home/zfs/x/zfs/include/sys/zfs_znode.h:164:
/usr/home/zfs/x/zfs/include/os/freebsd/zfs/sys/zfs_znode_impl.h:133:11: error: too few arguments to function call, expected 7, have 6
  133 |                 return (SET_ERROR(EIO));
      |                         ^~~~~~~~~~~~~~
/usr/home/zfs/x/zfs/include/os/freebsd/spl/sys/sdt.h:40:32: note: expanded from macro 'SET_ERROR'
   39 |         (*sdt_probe_func)(sdt_sdt___set__error->id, \
      |         ~~~~~~~~~~~~~~~~~
   40 |             (uintptr_t)err, 0, 0, 0, 0) : 0), err)
      |                                       ^
In file included from /usr/home/zfs/x/zfs/module/os/freebsd/spl/spl_policy.c:37:
/usr/home/zfs/x/zfs/include/sys/zfs_znode.h:225:11: error: too few arguments to function call, expected 7, have 6
  225 |                 return (SET_ERROR(EIO));
      |                         ^~~~~~~~~~~~~~
/usr/home/zfs/x/zfs/include/os/freebsd/spl/sys/sdt.h:40:32: note: expanded from macro 'SET_ERROR'
   39 |         (*sdt_probe_func)(sdt_sdt___set__error->id, \
      |         ~~~~~~~~~~~~~~~~~
   40 |             (uintptr_t)err, 0, 0, 0, 0) : 0), err)
      |                                       ^
2 errors generated.
*** Error code 1

Stop.
make[3]: stopped in /usr/home/zfs/x/zfs/module
gmake[2]: *** [Makefile:61: modules-FreeBSD] Error 1
gmake[2]: Leaving directory '/usr/home/zfs/x/zfs/module'
gmake[1]: *** [Makefile:12320: all-recursive] Error 1
gmake[1]: Leaving directory '/usr/home/zfs/x/zfs'
gmake: *** [Makefile:4648: all] Error 2
[zfs@vm1 ~/x/zfs]$ uname -a
FreeBSD vm1 15.0-CURRENT FreeBSD 15.0-CURRENT #0 main-n271031-cf2c79915a63: Thu Jul  4 09:44:29 UTC 2024     root@releng3.nyi.freebsd.org:/usr/obj/usr/src/amd64.amd64/sys/GENERIC amd64
```

Bulding master on FreeBSD 13 + FreeBSD 14  is fine.

**Comment by robn** [comment]
I think fixed by #16284.

**Comment by mcmilk** [comment]
> I think fixed by #16284.

Yes, thats LGTM.
